template/id.h.tmpl

```diff
@@ -44,6 +44,8 @@ enum ruby_id_types {
 #define ID_JUNK        RUBY_ID_JUNK
 #define ID_INTERNAL    RUBY_ID_INTERNAL
 
+#define ID_INTERNAL_P(id) ((id & RUBY_ID_SCOPE_MASK) == RUBY_ID_INTERNAL)
```

Comment:
I extracted this so at least it feels a bit less dirty to check for internal ids.

Reply:
Actually it already exists in `symbol.h` as `is_junk_id()`.